src/elements/content-sidebar/activity-feed/annotations/AnnotationActivity.js

```diff
@@ -81,9 +97,29 @@ const AnnotationActivity = ({
     const activityLinkMessage = isFileVersionUnavailable
         ? messages.annotationActivityVersionUnavailable
         : { ...linkMessage, values: { number: linkValue } };
+    const clickHandlers = useSuppressiveClicks<HTMLDivElement>({ onClick: handleOnSelect });
+    const interactiveDivProps = isInteractive
+        ? {
+              role: 'button',
+              tabIndex: '0',
+          }
+        : null;
+
+    React.useEffect(() => {
+        setIsInteractive(!(isMenuOpen || isEditing || isFileVersionUnavailable));
```

Comment:
It looks like this is derived state, which we usually want to avoid. Can we assign the result of these checks to a boolean value each render? Can we also clarify the conditions to avoid the negation?